.github/workflows/scripts/bump-versions.sh

```diff
@@ -0,0 +1,210 @@
+  version=${version#v}
+  if [[ ! $version =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+    echo "Invalid version: $version. Must be a semantic version (e.g., 1.2.3)."
+    exit 1
+  fi
+  eval "$var_name='$version'"
+}
+commit_changes=false
+create_pr=false
+# Parse named arguments
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    --current-beta-core) current_beta_core="$2"; shift ;;
+    --current-beta-contrib) current_beta_contrib="$2"; shift ;;
+    --current-stable) current_stable="$2"; shift ;;
+    --next-beta-core) next_beta_core="$2"; shift ;;
+    --next-beta-contrib) next_beta_contrib="$2"; shift ;;
+    --next-stable) next_stable="$2"; shift ;;
+    --commit) commit_changes=true ;;
+    --pull-request) create_pr=true ;;
+    *) echo "Unknown parameter passed: $1"; usage ;;
+  esac
+  shift
+done
+
+# Check if --pull-request is passed without --commit
+if [ "$create_pr" = true ] && [ "$commit_changes" = false ]; then
+  echo "--pull-request requires --commit"
+  usage
+fi
+
+# Check if at least one of the required arguments is provided
+if [ -z "$current_beta_core" ] && [ -z "$current_beta_contrib" ] && [ -z "$current_stable" ]; then
+  usage
+fi
+
+# Validate and strip versions
+if [ -n "$current_beta_core" ]; then
+  validate_and_strip_version current_beta_core
+fi
+if [ -n "$current_beta_contrib" ]; then
+  validate_and_strip_version current_beta_contrib
+fi
+if [ -n "$current_stable" ]; then
+  validate_and_strip_version current_stable
+fi
+if [ -n "$next_beta_core" ]; then
+  validate_and_strip_version next_beta_core
+fi
+if [ -n "$next_beta_contrib" ]; then
+  validate_and_strip_version next_beta_contrib
+fi
+if [ -n "$next_stable" ]; then
+  validate_and_strip_version next_stable
+fi
+
+# Function to bump the minor version and reset patch version to 0
+bump_version() {
+  local version=$1
+  local major
+  major=$(echo "$version" | cut -d. -f1)
+  local minor
+  minor=$(echo "$version" | cut -d. -f2)
+  local new_minor
+  new_minor=$((minor + 1))
+  echo "$major.$new_minor.0"
+}
+
+# Infer the next beta version if not supplied
+if  [ -n "$current_beta_core" ] && [ -z "$next_beta_core" ]; then
+  next_beta_core=$(bump_version "$current_beta_core")
+fi
+if  [ -n "$current_beta_contrib" ] && [ -z "$next_beta_contrib" ]; then
+  next_beta_contrib=$(bump_version "$current_beta_contrib")
+fi
+
+# Infer the next stable version if current_stable provided and next version not supplied
+if [ -n "$current_stable" ] && [ -z "$next_stable" ]; then
+  next_stable=$(bump_version "$current_stable")
+fi
+
+# List of files to update
+files=(
+  "distributions/otelcol-contrib/manifest.yaml"
+  "distributions/otelcol/manifest.yaml"
+  "distributions/otelcol-k8s/manifest.yaml"
+  "distributions/otelcol-otlp/manifest.yaml"
+  "Makefile"
+)
+
+# Update versions in each file
+for file in "${files[@]}"; do
+  if [ -f "$file" ]; then
+    temp_file=$(mktemp)
+    cp "$file" "$temp_file"
+    line_number=1
+    while IFS= read -r line; do
```

Comment:
Why does the script run `sed` for each line separately? Putting efficiency aside, this seems harder to understand than just running `sed` on the whole file(s). The replacement patterns are also less precise. For example, there are lines in `Makefile` that have `github.com/open-telemetry/opentelemetry-collector-contrib` in them, but are not subject to replacements.

I would imagine running the `sed` command on the whole files and specifying the replacement patterns specifically for each purpose. For example:
- to replace core versions in manifests: `sed "s/\(^.*go\.opentelemetry\.io\/collector\/.*\) v0\.111\.0/\1 v0.112.0/" distributions/otelcol/manifest.yaml`
- to replace `otelcol_version` in manifests: `sed "s/\(^\s\+otelcol_version:\) 0\.111\.0/\1 0.112.0/" distributions/otelcol/manifest.yaml`
- etc.